Kryptonite/Keypair+PGPFormat.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  Keypair+PGPFormat.swift
+//  Kryptonite
+//
+//  Created by Alex Grinman on 5/20/17.
+//  Copyright © 2017 KryptCo. All rights reserved.
+//
+
+import Foundation
+import Sodium
+import CommonCrypto
+import PGPFormat
+
+extension KeyType {
+    var pgpKeyType:PGPFormat.PublicKeyAlgorithm {
+        switch self {
+        case .Ed25519:
+            return PGPFormat.PublicKeyAlgorithm.ed25519
+        case .RSA:
+            return PGPFormat.PublicKeyAlgorithm.rsaSignOnly
+        }
+    }
+
+    func digestType(for hashAlgorithm:Signature.HashAlgorithm) -> DigestType {
+        switch self {
+        case .Ed25519:
+            return DigestType.ed25519
+        case .RSA:
+            switch hashAlgorithm {
+            case .sha1:
+                return DigestType.sha1
+            case .sha224:
+                return DigestType.sha224
+            case .sha256:
+                return DigestType.sha256
+            case .sha384:
+                return DigestType.sha384
+            case .sha512:
+                return DigestType.sha512
+            }
+        }
+    }
+}
+
+
+// MARK: PGPPublicKeyConvertible
+protocol PGPPublicKeyConvertible {
+    func pgpPublicKeyData() throws -> PGPFormat.PublicKeyData
+}
+
+struct UnknownPGPKeyFormat:Error {}
+struct ModulusTooShort:Error {}
+
+extension PublicKey {
+    func pgpPublicKey() throws -> PGPFormat.PublicKeyData {
+        guard let pgpKey = self as? PGPPublicKeyConvertible
+            else {
+                throw UnknownPGPKeyFormat()
+        }
+        
+        return try pgpKey.pgpPublicKeyData()
+    }
+}
+
+extension RSAPublicKey:PGPPublicKeyConvertible {
+    func pgpPublicKeyData() throws -> PGPFormat.PublicKeyData {
+        let (modulus, exponent) = try self.splitIntoComponents()
+        guard modulus.count >= 1 else {
+            throw ModulusTooShort()
+        }
+        
+        let modulusFixed = Data(bytes: modulus.bytes[1 ..< modulus.count])
+        
+        return PGPFormat.RSAPublicKey(modulus: modulusFixed, exponent: exponent)
+    }
+}
+
+extension Sign.PublicKey:PGPPublicKeyConvertible {
+    func pgpPublicKeyData() throws -> PGPFormat.PublicKeyData {
+        return PGPFormat.Ed25519PublicKey(rawData: self)
+    }
+}
+
+/** 
+    Extend Keypair Signatures to hash and return hash + signed hash
+ */
+
+extension KeyPair {
+    
+    /** 
+        Return the hash and the signed hash
+        Note: Ed25519 signs the hash itself as per OpenPGP RFC for Ed25519.
+     */
+    func sign(data:Data, using hashAlgorithm:Signature.HashAlgorithm) throws -> (hash:Data, signedHash:Data) {
+        var hash:Data
+        var signedHash:Data
+        
+        switch hashAlgorithm {
```

Comment:
I was thinking about it, but i only do it in one place so decided not to until we needed to replicate it